packages/docs/docs/docs/proxies.md

```diff
@@ -0,0 +1,150 @@
+
+A very important thing to note is that the code makes use of the EVM's `delegatecall` opcode which executes the callee's code in the context of the caller's state. That is, the logic contract controls the proxy's state and the logic contract's state is meaningless. Thus, the proxy doesn't only forward transactions to and from the logic contract, but also represents the pair's state. The state is in the proxy and the logic is in the particular implementation that the proxy points to.
+
+## Unstructured storage proxies
+
+A problem that quickly comes up when using proxies has to do with the way in which variables are stored in the proxy contract. Suppose that the proxy stores the logic contract's address in it's only variable `address public _implementation;`. Now, suppose that the logic contract is a basic token whose first variable is `address public _owner`. Both variables are 32 byte in size, and as far as the EVM knows, occupy the first slot of the resulting execution flow of a proxied call. When the logic contract writes to `_owner`, it does so in the scope of the proxy's state, and thus really writes to `_implementation`. This problem can be referred to as a "storage collision".
+
+|Proxy                     |Implementation           |
+|--------------------------|-------------------------|
+|address _implementation   |address _owner           | <=== Storage collision!
+|...                       |mapping _balances        |
+|                          |uint256 _supply          |
+|                          |...                      |
+
+There are many ways to overcome this problem, and the "unstructured storage" approach which ZeppelinOS implements works as follows. Instead of storing the `_implementation` address at the proxy's first storage slot, it chooses a pseudo random slot instead. This slot is sufficiently random, that the probability of a logic contract declaring a variable at the same slot is negligible. The same principle of randomizing slot positions in the proxy's storage is used in any other variables the proxy may have, such as an admin address (that is allowed to update the value of `_implementation`), etc.
+
+|Proxy                     |Implementation           |
+|--------------------------|-------------------------|
+|...                       |address _owner           |
+|...                       |mapping _balances        |
+|...                       |uint256 _supply          |
+|...                       |...                      |
+|...                       |                         |
+|...                       |                         |
+|...                       |                         |
+|...                       |                         |
+|address _implementation   |                         | <=== Randomized slot.
+|...                       |                         |
+|...                       |                         |
+   
+An example of how the randomized storage is achieved:
```

Comment:
I wouldn't use the word random, it sounds not-robust, I'd rather use something like `fixed`, explaining that we choose a specific position [making sure that Solidity won't ever use it](https://github.com/zeppelinos/labs/issues/82)